Google.Api.Gax.Grpc/ChannelPool.cs

```diff
@@ -44,8 +46,11 @@ public sealed class ChannelPool
         /// if they require any.
         /// </summary>
         /// <param name="scopes">The scopes to apply. Must not be null, and must not contain null references. May be empty.</param>
-        public ChannelPool(IEnumerable<string> scopes)
+        /// <param name="useUseJwtAccessWithScopes"></param>
+        public ChannelPool(IEnumerable<string> scopes, bool useUseJwtAccessWithScopes)
```

Comment:
Just to clarify my own thinking: we'd expect a generator change to call this constructor in a way that's consistent with what the client builder says about using JWTs.

I think this is fine *if* we don't allow a user to override. If we *do* allow a user to override the use of JWTs, we'd need to know not to use the channel pool in that case.

I think the implication of this is that the ClientBuilderBase property that I've proposing should be protected, virtual and read-only, probably *defaulting* to false (as otherwise old libraries taking a dependency on the new GAX would do the wrong thing) and being overridden in *every* generated library... and a separate `bool?` property for using JWTs that would be null by default, forcing use or non-use.

If none of that is clear, I can come up with another PR showing what I mean :)

Reply:
If the second bool? property is for allowing users to overwrite the behaviour and us to know not to use the pool if it has been set, then I agree this is a good solution. The same if we have one inmutable property only and users cannot decide wether to use the JWTs or not (I like it better if users can choose, but I'm fine with either).

An alternative that we might want to look at for a next major version is to maybe make useJwts (and scopes which would be the breaking bit) part of the key used to fetch a channel from the ChannelPool, that way we can facilitate reusing of channels for more use cases.

Reply:
Sorry if I don't get some details, my thoughts: 

I think we can keep things simple and NOT allow an override. We effectively have only GAPIC libraries in scope. If JWT is supported by the API, what would be the reason to override JWT? It is faster and more reliable. We can provide a way to override once the scope becomes broader than GAPIC libraries. 

Also, question, why we care about Pool is being set only in case of override? Or we always care? Then it means I miss something in current draft ? Something that disables pool in case of JWT or use JWT only if there is no pool ?

Reply:
> Also, question, why we care about Pool is being set only in case of override? Or we always care? Then it means I miss something in current draft ? Something that disables pool in case of JWT or use JWT only if there is no pool ?

The default pool uses very specific credentials, with the API default scopes, and even with your code as it is, not opted in to JWTs. So there are many scenarion in which we cannot use the pool, for instance, if the client provides specific scopes, or if the client provides a quota project, etc.